src/csharp/Pester/Factory.cs

```diff
@@ -50,6 +50,7 @@ public static ErrorRecord CreateErrorRecord(string errorId, string message, stri
                 ["Line"] = line,
                 ["LineText"] = lineText,
                 ["Terminating"] = terminating,
+                ["ShouldResult"] = shouldResult
```

Comment:
Should we set this when null? Not a change request, just looking to learn from others view on  scenarios like this. As it's just a dict key it wouldn't be breaking.

I doubt anyones using this factory method outside Pester, so 99% of the time `ShouldResult` will be provided.

Reply:
From a perf standpoint I don't think it makes that big of a difference, but there certainly can be a check if desired, that's not a big deal to implement.

Reply:
Was thinking more about user experience. If you'd prefer consistency over only relevant when ex printing `ErrorRecord.TargetObject` to screen. No worries, no change needed :)

Reply:
I made this conditional in 
34b4744